pydantic_evals/pydantic_evals/datasets.py

```diff
@@ -0,0 +1,376 @@
+                assessment_types.append(td)
+
+        params = cls._params()
+
+        class ClsDatasetRow(DatasetRow[params[0], params[1], params[2]]):
+            assessments: list[Union[tuple(assessment_types)]] = Field(default_factory=list)  # pyright: ignore  # noqa UP007
+
+        ClsDatasetRow.__name__ = cls.__name__ + 'Row'
+
+        class ClsDataset(cls):
+            rows: list[ClsDatasetRow]  # pyright: ignore
+
+        ClsDataset.__name__ = cls.__name__
+
+        return ClsDataset.model_json_schema()
+
+    # TODO: Task: Uncomment and finish implementing function to generate examples for a dataset using an LLM
+    # @classmethod
+    # def generate_dataset_examples(
+    #     cls,
+    #     model: models.Model | models.KnownModelName = 'gpt-4o',
+    #     min_count: int = 3,
+    #     dataset_path: Path | str = DEFAULT_DATASET_PATH,
+    # ):
+    #     dataset_path = cls._get_relative_path(dataset_path)
+    #     schema_ref = cls.generate_dataset_files(dataset_path=dataset_path, schema_path=None)
+    #
+    #     existing_content: str | None = None
+    #
+    #     try:
+    #         existing_rows = cls.from_yaml(dataset_path).rows
+    #         min_count = max(0, min_count - len(existing_rows))
+    #         if min_count == 0:
+    #             return  # nothing to do, already have enough examples
+    #         if existing_rows:
+    #             existing_content = dataset_path.read_text()
+    #     except FileNotFoundError:
+    #         pass  # in this case, we'll generate a new file, so we ignore the error
+    #
+    #     examples = asyncio.run(_generate_examples(cls, dataset_path, model, min_count))
+    #
+    #     if existing_content is None:
+    #         content = yaml.dump(to_jsonable_python(cls(rows=examples)), sort_keys=False)
+    #         content = _ensure_yaml_language_server_line(content, schema_ref)
+    #         dataset_path.write_text(content)
+    #     else:
+    #         new_lines = yaml.dump(to_jsonable_python(cls(rows=examples)), sort_keys=False).splitlines()
+    #         new_lines = new_lines[1:]  # drop the first line, which is the document start
+    #         new_content = _ensure_yaml_language_server_line(existing_content, schema_ref)
+    #         if not new_content.endswith('\n'):
+    #             new_content += '\n'
+    #         new_content += '\n'.join(new_lines)
+    #         dataset_path.write_text(new_content)
+
+    @classmethod
+    def _get_schema_path(cls, dataset_path: Path) -> Path:
+        return dataset_path.parent / f'./{dataset_path.stem}_schema.json'
+
+    @classmethod
+    def _get_relative_path(cls, path: Path | str) -> Path:
+        """Resolve relative paths as relative to the module in which the (sub)class is defined."""
+        path = Path(path)
+
+        if path.is_absolute():
+            return path
+
+        module_path = sys.modules[cls.__module__].__file__
+        if module_path == __file__:
```

Comment:
```suggestion
        if cls is Dataset:
```

Reply:
that doesn't work for `Dataset[MyType, MyOtherType, MyAnotherType]` because parametrized pydantic generics makes new classes.